posts/contributing-to-govuk-design-system.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Why I contribute to the GOV.UK Design System (and why you should too)
+date: 2022-05-03
+tags: post
+layout: post
+author:
+  name: Frankie Roberto
+  url: https://github.com/frankieroberto
+---
+
+Last week a new pattern was published in the GOV.UK Design System that helps users [confirm a phone number](https://design-system.service.gov.uk/patterns/confirm-a-phone-number/) using a security code sent by text message.
+
+I initiated this contribution almost two years ago when working on a service that was switching to using its own sign in system to avoid relying upon an off-the-shelf authentication service which was hard to maintain.
+
+As the service was quite sensitive, we decided to require a text message confirmation code when signing in – sometimes known as two factor authentication.
+
+The pattern of sending and requesting an SMS code is quite well established, used by many services in and out of government.
+
+However we still had a bunch of detailed design questions. Things like:
+
+* should we call it a security code, a verification code, or something else?
+* how long should the codes be valid for?
+* if a code doesn't arrive straight away and the user requests a new one, should we resend the same code or generate a new one?
+* how many digits should the code contain?
+* when first setting up an account, how we can we make sure the phone number is valid and can receive text messages without letting the user get stuck?
+
+For each of these questions, we looked at what existing services like GOV.UK Notify and HMRC did, and spoke to some of their designers to try to understand the reasons behind the decision they’d taken. Being able to reach out to the wider government design community like this, through platforms like the cross-government Slack and Twitter, is super valuable.
+
+There was also an existing thread on the GOV.UK Design System backlog for [identifying users](https://github.com/alphagov/govuk-design-system-backlog/issues/25), where designers and researchers had shared some screenshots and discussed the pattern. These [backlog threads](https://github.com/alphagov/govuk-design-system-backlog/issues) are another great resource, if sometimes hard to find.
+
+Once we had assimilated some of this existing work, plus made some design iterations of our own, we were ready to ship our feature. And once the dust had settled, I realised that writing this all up might be useful for other designers in the same situation.
```

Comment:
```suggestion
When we had assimilated some of this existing work, plus made some design iterations of our own, we were ready to ship. And once the dust had settled, I realised that writing this all up might be useful for other designers in the same situation.
```

Reply:
@paulrobertlloyd  how’s this, switching the first one to ‘when’ (as ‘once the dust had settled’ is idiomatic).

Also realised that “our feature” doesn’t add anything.

Reply:
👌